public/app/plugins/panel/heatmap/utils.ts

```diff
@@ -280,17 +302,31 @@ export function prepConfig(opts: PrepConfigOpts) {
                 }
               }
 
+              // For pre-bucketed data with explicit scale, calculate expansion factor from actual bucket spacing
+              // For calculated heatmaps, use the full log base
+              let expansionFactor: number = yExp;
+
+              if (yBucketScale !== undefined) {
+                // Try to infer the bucket factor from the actual data spacing
+                const yValues = u.data[1]?.[1];
+                if (Array.isArray(yValues) && yValues.length >= 2 && typeof yValues[0] === 'number') {
+                  expansionFactor = calculateBucketFactor(yValues, yExp);
+                }
+              }
```

Comment:
Calculate expansion from data spacing
Use calculateBucketFactor instead of fixed yExp